docs/guidelines/markup.rst

```diff
@@ -0,0 +1,170 @@
+
+
+Indendation
+-----------
+
+.. admonition:: Django
+    :class: `important`
+
+    - Always add an indend after django tags such as ``{% if %}``, ``{% forloop %}``, ``{% block %}`` ...
+    - Use singlelines within ``{% addtoblock %}`` for **files** and multilines for ``<code>``
+    - **Code readability** always wins
+
+Valid
+*****
+
+.. code-block:: django
+
+    {% block content %}
+        <div class="plugin-blog">
+            {% if true %}
+                <p>Hello World</p>
+            {% endif %}
+        </div>
+    {% endblock content %}
+
+    {% addtoblock "js" %}<script src="{% static "js/libs/jquery.min.js" %}"></script>{% endaddtoblock %}
+    {% addtoblock "js" %}
+    <script>
+    jQuery(document).ready(function ($) {
+        alert('hello world');
+    });
+    </script>
+    {% endaddtoblock %}
+
+Invalid
+*******
+
+.. code-block:: django
+
+    {% block content %}
+    <div class="plugin-blog">{% if true %}<p>Hello World</p>{% endif %}</div>
+    {% endblock content %}
+
+    {% addtoblock "js" %}
+    <script src="{% static "js/libs/jquery.min.js" %}"></script>
+    {% endaddtoblock %}
+    {% addtoblock "js" %}
+        <script>
+            jQuery(document).ready(function ($) {
+                alert('hello world');
+            });
+        </script>
+    {% endaddtoblock %}
+
+
+IDs vs Classes
+--------------
+
+.. admonition:: Don't use IDs
+    :class: `important`
+
+    - Avoid IDs ad all cost
+    - When using IDs always use a **unique name**
+
+You should **always** use classes instead if ID's. Classes represent a more OOP approach of adding and removing
+style sets like ``box box-wide box-hint``.
+
+Try to avoid declaring ID's at all. They should only be used to reference form elements or for in-page navigation
+in which case you need to make the name **absolutely unique**.
+
+Valid
+*****
+
+.. code-block:: html
+
+    <div class="box box-highlighted box-8723"> ... </div>
+    <!-- IDs only for navigation jumper through <a href="#page-anchor-team"></a> -->
+    <div id="page-anchor-team"></div>
+    <!-- IDs only for form elements -->
+    <label for="field-id12-firstname">Name</label>
+    <input type="text" name="firstname" id="field-id12-firstname">
+
+Invalid
+*******
+
+.. code-block:: html
+
+    <div class="box box-highlighted" id="box-8723"> ... </div>
+    <!-- IDs only for navigation jumper through <a href="#page-anchor-team"></a> -->
+    <div id="team"></div>
+    <!-- IDs only for form elements -->
+    <label for="firstname">Name</label>
+    <input type="text" name="firstname" id="firstname">
+
+
+Modular
+-------
+
+.. important::
+
+    Try to keep the html structure simple and avoid unnecessary elements. It is sometimes easier to use a single div with
+    a single class rather than multiple divs with multiple classes.
+
+For example, lets take a look at the following code snippet:
+
+.. code-block:: html
+
+    <div class="addon-blog">
+        <h2>My Blog</h2>
```

Comment:
this is not modular and will cause pain and suffering
in order to make things modular we should avoid type selectors at all costs
usually in big and long projects changes may come in between that would require to rewrite the code over and over
like change h2 to h3 based on the context in which addon is used for example
and then we suddenly have more selectors than we could handle, which already happens on some projects
```
.addon-blog
     h2.addon-blog-heading
     div.addon-blog-content
         p Hello world
```
would be zillion times easier to maintain and extend